README.md

```diff
@@ -16,91 +18,97 @@ This is an all-new API, so refer to [the `1.x` branch](https://github.com/schroc
 
 ## Quick Example
 
-Define authorization rules directly in the context module, in this case `MyApp.Blog`:
+Define authorization rules directly in the context module:
 
 ```elixir
+# lib/my_app/blog.ex
 defmodule MyApp.Blog do
-  @behaviour Bodyguard.Policy
+  use Bodyguard
 
-  # Implement this callback:
-  def authorize(:update_post, user, %{post: post}) do
+  # Bodyguard callback authorize/3
+  def authorize(:update_post, user, post) do
     # Return :ok or true to permit
     # Return :error, {:error, reason}, or false to deny
   end
-end
 
-# Authorize a controller action (note the keyword list is converted to a map for the callback):
-with :ok <- Bodyguard.permit(MyApp.Blog, :update_post, user, post: post) do
-  # ...
+  # This context function will return:
+  #   - {:ok, post} if successful
+  #   - {:error, changeset} if operation fails
+  #   - {:error, :unauthorized} if authorization fails
+  def update_post(user, post_id, post_params) do
+    post = MyApp.Repo.get!(MyApp.Blog.Post, post_id)
+    with :ok <- permit(:update_post, user, post) do  # <-- permit/3 helper function
+      do_update_post(post, post_params)
+    end
+  end
+
+  # Do the heavy lifting here
+  defp do_update_post(post, post_params) do
+    # ...
+  end
 end
 ```
 
-## Policies
+## Contexts
 
-To implement a policy, add `@behaviour Bodyguard.Policy` to a context, then define `authorize(action, user, params)` callbacks, which must return:
+Add `use Bodyguard` to a context, then define `authorize(action, user, params)` callbacks, which must return:
 
 * `:ok` or `true` to permit the action
 * `:error`, `{:error, reason}`, or `false` to deny the action
 
-The `action` argument, an atom, might map one-to-one with the actual context function name, or it can be more broad (e.g. `:manage_post` or `:read_post`) to indicate a rule encompassing a wider range of actions.
-
-```elixir
-defmodule MyApp.Blog do
-  @behaviour Bodyguard.Policy
+**Don't use these callbacks directly** – instead, use the three wrapper functions injected into the context: 
 
-  # Admin users can do anything
-  def authorize(_, %Blog.User{role: :admin}, _), do: true
+* `permit/3` to get an `:ok` or `{:error, reason}`
+* `permit?/3` to get a boolean
+* `permit!/3` to get an `:ok` or raise `Bodyguard.NotAuthorizedError`
```

Comment:
I was trying to copy the pattern that `Phoenix.View` uses for rendering views, where externally you use `Phoenix.View.render(module, template, params)` and it will delegate to the callback `render(template, params)` on the view, while doing convenient things like mapifying the keyword params. 

The difference, which I'm just now realizing, is that these Bodyguard callbacks are meant to be used within the module itself, while the Phoenix callbacks are meant to be used externally. That's a fundamental difference which maybe makes this design not work as well.

I'm trying to keep everything piped through the public `Bodyguard.permit/4` API in case it gains more functionality later. I couldn't do that with `authorize/3` because it's already defined on the context, hence the use of the different name `permit`.

That's why I'm kind of stuck. Maybe if I just "give up" on mapifying params (how often is that needed, really? Usually you're probably just passing in a single param to authorize, and if you're not just pass in a map directly) then using `authorize/!/?` everywhere could work.